From c68ed3899afee2e25da2eec6109076b2577e1b76 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sat, 5 Sep 2026 17:11:44 -0500
Subject: [PATCH] Revert "HACK: win32u: Implement window move hack"

This reverts commit 6f7463d364b1b7c99b276c258de9fcff16de6f3b.
---
 dlls/win32u/window.c | 64 +-------------------------------------------
 1 file changed, 1 insertion(+), 63 deletions(-)

diff --git a/dlls/win32u/window.c b/dlls/win32u/window.c
index fb751b2f879..744244a4e17 100644
--- a/dlls/win32u/window.c
+++ b/dlls/win32u/window.c
@@ -2218,19 +2218,6 @@ static BOOL is_fullscreen( const MONITORINFO *info, const RECT *rect )
            rect->top <= info->rcMonitor.top && rect->bottom >= info->rcMonitor.bottom;
 }
 
-static int use_move_hack(void)
-{
-    static volatile int enabled = -1;
-
-    if (enabled == -1)
-    {
-        const char *env = getenv("WINE_MOVE_HACK");
-        enabled = (env && !strcmp(env, "1"));
-    }
-
-    return enabled;
-}
-
 /***********************************************************************
  *           apply_window_pos
  *
@@ -2244,61 +2231,12 @@ static BOOL apply_window_pos( HWND hwnd, HWND insert_after, UINT swp_flags, stru
     HWND owner_hint, surface_win = 0, toplevel;
     UINT raw_dpi, monitor_dpi, dpi = get_thread_dpi();
     BOOL ret, is_layered, is_child, need_icons = FALSE;
-    struct window_rects old_rects, adjusted;
+    struct window_rects old_rects;
     RECT extra_rects[3];
     struct window_surface *old_surface;
     HICON icon, icon_small;
     ICONINFO ii, ii_small;
 
-    /* HACK: move windows within the virtual screen on winewayland */
-    if (use_move_hack())
-    {
-        RECT temp, *adj;
-        RECT virtual_screen = get_virtual_screen_rect( get_thread_dpi(), MDT_DEFAULT );
-
-        adjusted = *new_rects;
-        adj = &adjusted.client;
-
-        intersect_rect(&temp, &virtual_screen, adj);
-
-        /* we aren't off screen */
-        if (!IsRectEmpty(&temp))
-        {
-            LONG offset_x = 0, offset_y = 0;
-
-            if (adj->bottom - adj->top <=
-                virtual_screen.bottom - virtual_screen.top)
-            {
-                if (adj->bottom > virtual_screen.bottom)
-                    offset_y = virtual_screen.bottom - adj->bottom;
-                else if (virtual_screen.top > adj->top)
-                    offset_y = virtual_screen.top - adj->top;
-            }
-
-            if (adj->right - adj->left <=
-                virtual_screen.right - virtual_screen.left)
-            {
-                if (adj->right > virtual_screen.right)
-                    offset_x = virtual_screen.right - adj->right;
-                else if (virtual_screen.left > adj->left)
-                    offset_x = virtual_screen.left - adj->left;
-            }
-
-            OffsetRect(&adjusted.client, offset_x, offset_y);
-            OffsetRect(&adjusted.visible, offset_x, offset_y);
-            OffsetRect(&adjusted.window, offset_x, offset_y);
-
-            if (offset_x != 0 || offset_y != 0)
-            {
-                TRACE("vscreen rect %s\n", wine_dbgstr_rect(&virtual_screen));
-                TRACE("Original window rects: %s\n", debugstr_window_rects(new_rects));
-                TRACE("Adjusted window rects: %s\n", debugstr_window_rects(&adjusted));
-
-                new_rects = &adjusted;
-            }
-        }
-    }
-
     toplevel = NtUserGetAncestor( hwnd, GA_ROOT );
     is_layered = new_surface && new_surface->alpha_mask;
     is_child = toplevel && toplevel != hwnd;
